test/e2e/performanceprofile/functests/1_performance/cpu_management.go

```diff
@@ -664,13 +664,14 @@ func checkPodHTSiblings(testpod *corev1.Pod) bool {
 		"-c",
 		fmt.Sprintf("/bin/crictl inspect %s | /bin/jq -r '.info.runtimeSpec.linux.resources.cpu.cpus'", containerID),
 	}
-	output, err := nodes.ExecCommandOnNode(cmd, workerRTNode)
+	node, err := nodes.GetByName(testpod.Spec.NodeName)
```

Comment:
this is indeed more correct (and IMHO easier to follow too). I'm just wondering in which circumnstances the `workerRTNode` can be different from `testpod.Spec.NodeName`

Reply:
We are not telling the `testpod` on which node we want it, so it can land on a node which is different than the `workerRTNode`.
But this PR actually exposed the real problem:
```
  [FAILED] failed to get node ""
  Unexpected error:
      <*errors.errorString | 0xc0008acb30>: {
          s: "failed to get node for the node \"\"",
      }
      failed to get node for the node ""
  occurred
  In [It] at: /go/src/github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/1_performance/cpu_management.go:668 @ 07/24/23 10:36:40.373```

seems like our pod is pending/failed to get scheduled for some reason because `testpod.Spec.NodeName` is empty

Reply:
My bad, we do set the nodeSelector so the original fix doesn't help.
So we know the pod is going to the correct node, but for some reason it failed to get admitted. i'll keep digging.

Reply:
no problem